From 6d47f390242eb63f28e637cb1a8ebb24ab95cf5a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lothar=20Wa=C3=9Fmann?= <LW@KARO-electronics.de>
Date: Mon, 18 Mar 2019 15:37:04 +0100
Subject: [PATCH 1/8] drm/panel: simple: Add support for Tianma TM101JVHG32
 panel

This patch adds support for the Tianma 10.1" TM101JVHG32 TFT panel
with LVDS interface.
---
 .../display/panel/tianma,tm101jvhg32.txt      |  7 ++++
 drivers/gpu/drm/panel/panel-simple.c          | 33 +++++++++++++++++++
 2 files changed, 40 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/display/panel/tianma,tm101jvhg32.txt

diff --git a/Documentation/devicetree/bindings/display/panel/tianma,tm101jvhg32.txt b/Documentation/devicetree/bindings/display/panel/tianma,tm101jvhg32.txt
new file mode 100644
index 000000000000..69aadf85f021
--- /dev/null
+++ b/Documentation/devicetree/bindings/display/panel/tianma,tm101jvhg32.txt
@@ -0,0 +1,7 @@
+Tianma Micro-electronics TM101JVHG32 10.1" WUXGA TFT LCD panel
+
+Required properties:
+- compatible: should be "tianma,tm101jvhg32"
+
+This binding is compatible with the simple-panel binding, which is specified
+in simple-panel.txt in this directory.
diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index a76423057873..5c240d26de8b 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -2368,6 +2368,36 @@ static const struct panel_desc_dsi panasonic_vvx10f004b00 = {
 	.lanes = 4,
 };
 
+static const struct display_timing tianma_tm101jvhg32_timing = {
+	.pixelclock = { 62600000, 68200000, 78100000, },
+	.hactive = { 1280, 1280, 1280, },
+	.hfront_porch = { 15, 64, 159, },
+	.hback_porch = { 5, 4, 100, },
+	.hsync_len = { 20, 1, 164, },
+	.vactive = { 800, 800, 800, },
+	.vfront_porch = { 3, 40, 99, },
+	.vback_porch = { 2, 1, 61, },
+	.vsync_len = { 1, 1, 128, },
+	.flags = DISPLAY_FLAGS_DE_HIGH |
+		DISPLAY_FLAGS_HSYNC_LOW | DISPLAY_FLAGS_VSYNC_LOW,
+};
+
+static const struct panel_desc_dsi tianma_tm101jvhg32 = {
+	.desc = {
+		.timings = &tianma_tm101jvhg32_timing,
+		.num_timings = 1,
+		.bpc = 8,
+		.size = {
+			.width = 217,
+			.height = 136,
+		},
+		.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
+	},
+	.flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_SYNC_PULSE,
+	.format = MIPI_DSI_FMT_RGB888,
+	.lanes = 4,
+};
+
 static const struct of_device_id dsi_of_match[] = {
 	{
 		.compatible = "auo,b080uan01",
@@ -2384,6 +2414,9 @@ static const struct of_device_id dsi_of_match[] = {
 	}, {
 		.compatible = "panasonic,vvx10f004b00",
 		.data = &panasonic_vvx10f004b00
+	}, {
+		.compatible = "tianma,tm101jvhg32",
+		.data = &tianma_tm101jvhg32,
 	}, {
 		/* sentinel */
 	}
-- 
2.20.1

